File: c/002.c
#include <stdio.h>

int fibonacci(unsigned int n)
{
	return (n == 0) || (n == 1)
		? n
		: fibonacci(n - 1) + fibonacci(n - 2);
}

int main(void)
{
	unsigned int fibo_term = 0;
	unsigned int sum = 0;
	unsigned int n = 0;

	while ((fibo_term = fibonacci(n++)) <= 4000000)
		if (fibo_term % 2 == 0)
			sum += fibo_term;
	
	printf("sum = %d\n", sum);

	return 0;
}

File: haskell/002.hs
-- Project Euler
-- Problem 2: Even Fibonacci numbers

fib :: Integral a => (a, a) -> (a, a)
fib (a, b) = (b, a + b)

main = print
     . sum
     . filter even
     . takeWhile (< 4000000)
     . map fst
     $ iterate fib (0, 1)

